phper/src/builder/function.rs

```diff
@@ -0,0 +1,74 @@
+use crate::builder::argument::Argument;
+use crate::builder::data_type::AtomicReturnType;
+use crate::builder::data_type::AtomicType;
+use crate::builder::describe::Describe;
+
+pub struct Function {
+    pub name: String,
+    pub arguments: Vec<Argument>,
+    pub returns: AtomicReturnType,
+}
+
+impl Function {
+    pub fn new(name: String) -> Function {
+        Function {
+            name,
+            arguments: vec![],
+            returns: AtomicReturnType::Of(AtomicType::Mixed),
+        }
+    }
+
+    pub fn with_argument(mut self, argument: Argument) -> Self {
+        self.arguments.push(argument);
+
+        self
+    }
+
+    pub fn returns(mut self, return_type: AtomicReturnType) -> Self {
+        self.returns = return_type;
+
+        self
+    }
+}
+
+/// Describe `Function`.
+///
+/// Example:
+///
+/// ```rust
+/// use phper::builder::argument::Argument;
+/// use phper::builder::data_type::AtomicReturnType;
+/// use phper::builder::data_type::AtomicType;
+/// use phper::builder::describe::Describe;
+/// use phper::builder::function::Function;
+///
+/// let function = Function::new("format".to_owned())
+///     .with_argument(Argument::new("message", AtomicType::String))
+///     .with_argument(
+///         Argument::new(
+///             "args",
+///             AtomicType::Union(vec![AtomicType::String, AtomicType::Int, AtomicType::Float]),
+///         )
+///         .varadic(),
+///     )
+///     .returns(AtomicReturnType::Of(AtomicType::String));
+///
+/// assert_eq!(
+///     "function format(string $message, string|int|float ...$args): string { exit(1); }",
```

Comment:
`exit(1)` is not needed in stubs, are they are not parsed by PHP, but only by static analysis tools and editors, however, adding it makes the stub not generate any errors when parsed by PHP ( see https://3v4l.org/PmeGW )